### PR TITLE
Add migration for checksums to collections.KeySet

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```text
feat: add migration for checksums to collections.KeySet
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/keeper/migrations.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+// NewMigrator returns a new Migrator.
+func NewMigrator(keeper Keeper) Migrator {
+	return Migrator{
+		keeper: keeper,
+	}
+}
+
+// MigrateChecksums migrates the wasm store from using a single key to
+// store a list of checksums to using a KeySet to store the checksums.
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/keeper/migrations.go:23]: just to add some more info on where `KeySet` is coming from 

```suggestion
// store a list of checksums to using a collections.KeySet to store the checksums.
```

File: modules/light-clients/08-wasm/keeper/migrations.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+	return hashes.Checksums, nil
+}
+
+// deleteChecksums deletes the checksums stored under the KeyChecksums key.
+func (m Migrator) deleteChecksums(ctx sdk.Context) error {
+	store := m.keeper.storeService.OpenKVStore(ctx)
+	err := store.Delete([]byte(types.KeyChecksums))
+
+	return err
```
Comment (damiannolan) [inline modules/light-clients/08-wasm/keeper/migrations.go:72]: could `return store.Delete()` but 🤷🏻

**Review by damiannolan — APPROVED** [review]
Excellent work! I left only minor nits, all looks good to me from what I can tell and nice tests too! 

LGTM and make an e2e for this as well!

File: modules/light-clients/08-wasm/types/keys.go [hunk 3]
```diff
@@ -9,4 +9,7 @@ const (
 
 	// Wasm is the client type for IBC light clients created using 08-wasm
 	Wasm = ModuleName
+
+	// KeyChecksums is the key under which all checksums are stored
+	KeyChecksums = "checksums"
```
Comment (colin-axner) [inline modules/light-clients/08-wasm/types/keys.go:14]: should we mark this as deprecated? or add a note that it was deleted in the upgrade to v8?

File: proto/ibc/lightclients/wasm/v1/wasm.proto [hunk 4]
```diff
@@ -31,3 +31,8 @@ message ClientMessage {
 
   bytes data = 1;
 }
+
+// Checksums defines a list of all checksums that are stored
+message Checksums {
+  repeated bytes checksums = 1;
+}
```
Comment (colin-axner) [inline proto/ibc/lightclients/wasm/v1/wasm.proto:38]: maybe mark as deprecated?

File: modules/light-clients/08-wasm/keeper/migrations.go (hunk 2 above)
Comment (damiannolan) [reply modules/light-clients/08-wasm/keeper/migrations.go:72]: :D

**Review by colin-axner — APPROVED** [review]
Simple and clean. Great work :raised_hands:

File: modules/light-clients/08-wasm/types/keys.go (hunk 3 above)
Comment (damiannolan) [reply modules/light-clients/08-wasm/types/keys.go:14]: we could put into the migrations file?

Comment (damiannolan) [reply modules/light-clients/08-wasm/types/keys.go]: either works, I think its fine to leave where it is if its marked as deprecated now :)

Comment (colin-axner) [reply modules/light-clients/08-wasm/types/keys.go]: I thought about that as well. Happy with either approach

**Pull request merged**